server/athenian/api/kms.py

```diff
@@ -0,0 +1,49 @@
+import base64
+import io
+import logging
+import os
+from typing import Union
+
+from gcloud.aio.kms import encode, KMS
+
+from athenian.api import metadata
+
+
+class AthenianKMS:
+    """Google Cloud Key Management Service for Athenian API."""
+
+    log = logging.getLogger("%s.kms" % metadata.__package__)
+
+    def __init__(self):
+        """Initialize a new instance of AthenianKMS class."""
+        evars = {}
+        for var, env_name in (("keyproject", "GOOGLE_KMS_PROJECT"),
+                              ("keyring", "GOOGLE_KMS_KEYRING"),
+                              ("keyname", "GOOGLE_KMS_KEYNAME")):
+            evars[var] = x = os.getenv(env_name)
+            if x is None:
+                raise EnvironmentError(
+                    "%s must be defined, see https://cloud.google.com/kms/docs/reference/rest"
+                    % env_name)
+        service_file_inline = os.getenv("GOOGLE_KMS_SERVICE_ACCOUNT_JSON_INLINE")
+        if service_file_inline is not None:
+            service_file = io.StringIO(service_file_inline)
+        else:
+            service_file = os.getenv("GOOGLE_KMS_SERVICE_ACCOUNT_JSON")
+        self._kms = KMS(**evars, service_file=service_file)
+        self.log.info("Using Google KMS %(keyproject)s/%(keyring)s/%(keyname)s" % evars)
+
+    async def encrypt(self, plaintext: Union[bytes, str]) -> str:
+        """Encrypt text using Google KMS."""
+        return await self._kms.encrypt(encode(plaintext))
+
+    async def decrypt(self, ciphertext: str) -> bytes:
+        """Decrypt text using Google KMS."""
+        # we cannot use gcloud.aio.kms.decode because it converts bytes to string with str.decode()
```

Comment:
Hmm, that's a good point, and we should probably not be enforcing that type requirement on users, especially for what is basically just a convenience method. Will look into it :)

Reply:
Thank you! This seemed to little feature to raise an issue for :smile: